core/message-handling_test.go

```diff
@@ -275,8 +275,9 @@ func TestMakeReplicaMessageProcessor(t *testing.T) {
 		prepare := messageImpl.NewPrepare(primary, viewForPrimary(n, primary), request)
 
 		mock.On("messageProcessor", request).Return(false, fmt.Errorf("Error")).Once()
+		mock.On("uiMessageProcessor", prepare).Return(true, nil).Once()
 		_, err = process(prepare)
-		assert.Error(t, err, "Failed to process embedded Request")
+		assert.NoError(t, err)
```

Comment:
It is my understanding that we will get a warning log at this line but it doesn't matter whether the log is generated or not (i.e. there is an ignored message or not) in this test case.

Reply:
Exactly, this test does only check protocol logic and ignores diagnostic output.